docs/proposals/role-entry-points.md

```diff
@@ -0,0 +1,302 @@
+# Entry-point based roles
+
+## Variables inside a role are too open
+
+Each role consists of a set of tasks and variables read/set in some of the tasks.
+Some of the tasks can store their results into registered variables,
+e.g. store stdout of a docker pull command.
+Some of the tasks may be run only under certain conditions that are generated
```

Comment:
There's lots of things dynamically generated, like 'openshift.common.x'.  Should we consume openshift.common.x?  No, we shouldn't, but the way we are currently using openshift_facts kind of forces our hand.